lib/fizzy/execute.cpp

```diff
@@ -316,7 +317,8 @@ template <typename DstT>
 inline bool store_into_memory(bytes& memory, Stack<uint64_t>& stack, const uint8_t*& immediates)
 {
     const auto value = static_cast<DstT>(stack.pop());
-    const auto address = static_cast<uint32_t>(stack.pop());
+    // Addressing is 32-bit, but we keep the value as 64-bit to detect overflows.
+    const auto address = stack.pop() & 0xffffffff;
     // NOTE: alignment is dropped by the parser
     const auto offset = read<uint32_t>(immediates);
     if ((address + offset + sizeof(DstT)) > memory.size())
```

Comment:
I think this is also bad, because we should check both `address+offset` is not overflowing and that `address+offset+sizeof` is no exceeding the memory.